config/initializers/redis_hacks.rb

```diff
@@ -4,34 +4,36 @@
 # See https://github.com/redis/redis-rb/pull/856
 # We can remove this once we can upgrade the gem to >= v4.1.2
 
-module RedisHacks
-  def sentinel_detect
-    @sentinels.each do |sentinel|
-      client = Redis::Client.new(@options.merge({
-        :host => sentinel[:host],
-        :port => sentinel[:port],
-        :password => sentinel[:password], # https://github.com/redis/redis-rb/pull/856
-        :reconnect_attempts => 0,
-      }))
+class Redis
+  class Client
+    class Connector
+      class Sentinel < Connector
+        def sentinel_detect
+          @sentinels.each do |sentinel|
+            client = Client.new(@options.merge({
+              :host => sentinel[:host],
+              :port => sentinel[:port],
+              :password => sentinel[:password], # https://github.com/redis/redis-rb/pull/856
+              :reconnect_attempts => 0,
+            }))
 
-      begin
-        if result = yield(client)
-          # This sentinel responded. Make sure we ask it first next time.
-          @sentinels.delete(sentinel)
-          @sentinels.unshift(sentinel)
+            begin
+              if result = yield(client)
+                # This sentinel responded. Make sure we ask it first next time.
+                @sentinels.delete(sentinel)
+                @sentinels.unshift(sentinel)
 
-          return result
+                return result
+              end
+            rescue BaseConnectionError
```

Comment:
Please rebase so we can preserve the changes from https://github.com/3scale/porta/pull/1700.

Reply:
I thought from the commentary that it should just be reverted.
Isn't it the case @guicassolato ? 
https://github.com/3scale/porta/pull/1700/files#diff-699d2daa12e69e85956ec761bec65cb7R5

Reply:
Or we can just remove this hack ?

Reply:
We cannot remove it, @hallelujah. If had asked me yesterday, I'd said otherwise. But then we merged https://github.com/3scale/porta/pull/1700 which fixes a problem in the same method that we are hacking and that is not fixed even in the latest version of the gem. In other words, we still need the hack.